R/prepare_weo_2024_scenario.R

```diff
@@ -6,35 +6,45 @@
 #'   `WEO2024_Extended_Data_World.csv` import.
 #' @param weo_2024_fig_chptr_3_raw A tidyxl data frame containing a raw import
 #'   of `WEO2024_Figures_Chapter_03.xlsx`.
-#' @param iea_global_ev_raw A data frame containing a raw `IEA Global EV Data
-#'   2024.csv` import.
+#' @param iea_global_ev_raw A data frame containing a raw import of 'electric-vehicle-sales-by-region-and-scenario-2030-and-2035.xlsx'
+#' sheet : electric-vehicle-sales-by-regio
 #' @param mpp_ats_raw A tidyxl data frame containing a raw import of `2022-08-12
 #'   - MPP ATS - RPK and GHG intensity.xlsx`.
-#'
+#' @param hybrid_methodology A character chain explaining how to handle Hybrid Vehicles
+#' @param iea_sales_share_ev A data frame containing a raw import of 'electric-vehicle-sales-by-region-and-scenario-2030-and-2035.xlsx'
+#' - sheet: electric vehicle share-ev
+#' @param iea_sales_share_ev A data frame containing a raw import of 'electric-vehicle-sales-by-region-and-scenario-2030-and-2035.xlsx'
+#' - sheet: electric-vehicle-share-bev-phev
 #' @return A prepared WEO 2024 scenario data-frame.
 #'
 #' @importFrom dplyr %>%
 #'
 #' @export
-
 prepare_weo_2024_scenario <- function(weo_2024_ext_data_regions_raw,
                                       weo_2024_ext_data_world_raw,
                                       weo_2024_fig_chptr_3_raw,
-                                      iea_global_ev_raw,
+                                      iea_global_ev_2024_raw,
+                                      iea_sales_share_ev,
+                                      iea_sales_share_bev_phev,
+                                      hybrid_methodology,
                                       mpp_ats_raw) {
-  weo_2024_automotive <- weo_2024_extract_automotive(weo_2024_fig_chptr_3_raw, iea_global_ev_raw)
+  weo_2024_automotive <- weo_2024_extract_automotive(
+    iea_global_ev_2024_raw,
+    iea_sales_share_ev,
+    iea_sales_share_bev_phev,
+    hybrid_methodology)
   weo_2024_aviation <- weo_2024_extract_aviation(mpp_ats_raw, weo_2024_ext_data_world_raw)
   weo_2024_fossil_fuels <- weo_2024_extract_fossil_fuels(weo_2024_fig_chptr_3_raw)
   weo_2024_power <- weo_2024_extract_power(weo_2024_ext_data_regions_raw, weo_2024_ext_data_world_raw)
-  weo_2024_steel_cement <- weo_2024_extract_steel_cement(weo_2024_ext_data_world_raw, weo_2024_fig_chptr_3_raw)
+  #weo_2024_steel_cement <- weo_2024_extract_steel_cement(weo_2024_ext_data_world_raw, weo_2024_fig_chptr_3_raw)
```

Comment:
before merging to `main`, commented code should be removed, or if there's a really good reason to keep it a comment explaining why it is left in but commented out should be added

Reply:
@Antoine-Lalechere I think the feedback that we received indicates that steel and cement can be removed for now?